dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -170,14 +164,7 @@ public Entry createAndSetDiscourseTopic(Long id, Integer categoryId) {
         }
 
         // Verify and set category
-        Integer category = defaultDiscourseCategoryId;
-        if (categoryId != null) {
-            if (categoryId.equals(testDiscourseCategoryId) || categoryId.equals(defaultDiscourseCategoryId)) {
-                category = categoryId;
-            } else {
-                throw new CustomWebApplicationException("Category " + categoryId + " is not a valid category.", HttpStatus.SC_BAD_REQUEST);
-            }
-        }
+        Integer category = Integer.parseInt(discourseCategoryId);
```

Comment:
See my comment in DockstoreWebserviceConfiguration.java. If you don't want to go with that approach, I would suggest moving this up to line 97, so that you only do it once. Also you should catch NumberFormatException or errors can occur.